### ISSUE TITLE
Npm person can be a string

### ISSUE THREAD

**Issue by krassowski** [agoose77/hatch-nodejs-version#19]
Npm person can be a string as per https://docs.npmjs.com/cli/v9/configuring-npm/package-json#people-fields-author-contributors, e.g. 

```js
{
  "author": "Barney Rubble <b@rubble.com> (http://barnyrubble.tumblr.com/)"
}
```

is equivalent to

```js
{
  "author": {
     "name" : "Barney Rubble",
    "email" : "b@rubble.com",
    "url" : "http://barnyrubble.tumblr.com/"
   }
}
```

But currently if the string is use `hatch_nodejs_version` errors out with:

```
        File "/tmp/pip-build-env-44au558q/overlay/lib/python3.10/site-packages/hatch_nodejs_version/metadata_source.py", line 176, in update
          authors = [self._parse_person(package["author"])]
        File "/tmp/pip-build-env-44au558q/overlay/lib/python3.10/site-packages/hatch_nodejs_version/metadata_source.py", line 140, in _parse_person
          if {"url", "email"} & person.keys():
      AttributeError: 'str' object has no attribute 'keys'
```

**Comment by krassowski** [comment]
Hmm, though specification seems to suggest it is always an object

>  A "person" is an object with a "name" field and optionally "url" and "email", like this:

But JSON schema (and real life examples) show it accepts just a string:

**Comment by krassowski** [comment]
Over in https://github.com/jupyterlab/jupyterlab/issues/14963#issuecomment-1675423316 @vikramaditya91 brought up a nice example of yarn explicitly allowing either way of defining an author.

**Comment by agoose77** [comment]
Yeah this is confusing - I also read the spec requires an object. Yet, it seems that the wild has packages that don't satisfy this rule.

We should be permissive in parsing this, so we should create a pr.

**Comment by agoose77** [comment]
It's not clear to me exactly why we don't already support this, but I think I misread the spec. As such, I'm happy to merge this PR — thank you for contributing!

**Comment by agoose77** [comment]
Fixed by #20
